src/components/log-in/log-in.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import './log-in.scss';
+import btoa from 'btoa';
+import { setToken } from '../../redux/actions.js';
+import { connect } from 'react-redux';
+
+class Login extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      username: '',
+      email: '',
+      password: '',
+      url: process.env.REACT_APP_END_POINT_URL,
+      userExists: false
+    }
+    this.handleSignup = this.handleSignup.bind(this);
+    this.handleLogin = this.handleLogin.bind(this);
+  }
+
+  handleSignup (e) {
+    e.preventDefault();
+    fetch(`${this.state.url}/sign-up`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(
+        {
+          "username": this.state.username,
+          "email": this.state.email,
+          "password": this.state.password
+        }
+      )
+      })
+     .then(res => res.json())
+     .then(res => {
+      localStorage.setItem('token', res.token);
+      this.props.setToken(res.token)
+      this.props.close()
+     })
+
+  }
+
+  handleLogin = async (e, res) => {
+    e.preventDefault();
+    const payload = btoa(`${this.state.email}:${this.state.password}`);
+    console.log(payload);
+    fetch(`${this.state.url}/log-in`, {
+      method: 'GET',
+      headers: {
+        'Authorization': 'Basic ' + payload,
+      },
+      })
+      .then(res => res.json())
+      .then(res => {
+        localStorage.setItem('token', res.token);
```

Comment:
would recommend removing "localStorage.setItem('token', res.token);" since you have "   this.props.setToken(res.token)"